Phase8/EFCore/WebApi/Controllers/SearchController.cs

```diff
@@ -0,0 +1,40 @@
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Mvc;
+using WebApi.Services;
+
+namespace WebApi.Controllers
+{
+    [Route("[controller]/[action]")]
+    public class SearchController : ControllerBase
+    {
+        private readonly IInvertedIndexService _invertedIndex;
+
+        public SearchController(IInvertedIndexService invertedIndex)
+        {
+            _invertedIndex = invertedIndex;
+        }
+
+        [HttpGet]
+        public List<string> Query(string query)
+        {
+            return _invertedIndex.Query(query);
+        }
+
+        [HttpPost]
+        public IActionResult PostDocuments([FromBody] Dictionary<string, string> fileContents)
+        {
+            _invertedIndex.AddDocuments(fileContents);
+            return Ok(new
+            {
+                fileContents
+            });
+        }
```

Comment:
users are probably not supposed to send the documents to server over http request, but they only ask for search results from server (imagine sending a ton of documents over internet!).

Reply:
you're right. Actual documents should be added with ConsoleApp to the database but users can also add some documents to the available documents